projects/packages/masterbar/src/admin-menu/class-base-admin-menu.php

```diff
@@ -580,12 +587,16 @@ public function dashboard_switcher_scripts() {
 				var viewLink = $( '#view-link' );
 				var viewWrap = $( '#view-wrap' );
 
-				viewLink.on( 'click', function() {
+				// Remove any existing click handlers before adding a new one
+				viewLink.off( 'click' ).on( 'click', function() {
```

Comment:
Non-blocking question: Rather than removing events, cannot we use another static variable here too?

```
public function dashboard_switcher_scripts() {
	static $is_script_added = false;
	if ( $is_script_added ) {
		return;
	}

	wp_add_inline_script( ... );
	$is_script_added = true;
}
```

Reply:
Yeah, that's much cleaner, you're right! Done in d3eb69a12d4564aeeeccea437e54e6890df0edd5